File: medplum-java/medplum-java-model/src/main/java/com/fhir/MedicationKnowledge_Kinetics.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMedicationKnowledge_Kinetics.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface MedicationKnowledge_Kinetics {
  Optional<List<Extension>> extension();

  Optional<List<Quantity>> areaUnderCurve();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Quantity>> lethalDose50();

  Optional<Duration> halfLifePeriod();

  Optional<String> id();

  static ImmutableMedicationKnowledge_Kinetics.Builder builder() {
    return ImmutableMedicationKnowledge_Kinetics.builder();
  }
}

File: medplum-typescript/src/model/fhir/MedicationKnowledge_Kinetics.ts
import { Duration } from "./Duration";
import { Extension } from "./Extension";
import { Quantity } from "./Quantity";

export interface MedicationKnowledge_Kinetics {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    areaUnderCurve: Quantity[] | null | undefined;
    lethalDose50: Quantity[] | null | undefined;
    halfLifePeriod: Duration | null | undefined;
}
